lib/src/main/java/uk/gov/di/ipv/cri/address/library/service/CredentialIssuerService.java

```diff
@@ -0,0 +1,84 @@
+package uk.gov.di.ipv.cri.address.library.service;
+
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.utils.URLEncodedUtils;
+import software.amazon.awssdk.services.dynamodb.model.DynamoDbException;
+import uk.gov.di.ipv.cri.address.library.error.ErrorResponse;
+import uk.gov.di.ipv.cri.address.library.exception.CredentialRequestException;
+import uk.gov.di.ipv.cri.address.library.helpers.ListUtil;
+import uk.gov.di.ipv.cri.address.library.models.CanonicalAddressWithResidency;
+import uk.gov.di.ipv.cri.address.library.persistence.DataStore;
+import uk.gov.di.ipv.cri.address.library.persistence.item.AddressSessionItem;
+
+import java.nio.charset.Charset;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+public class CredentialIssuerService {
+    public static final String AUTHORIZATION_HEADER_KEY = "Authorization";
+    public static final String SUB = "sub";
+    private final DataStore<AddressSessionItem> dataStore;
+
+    public CredentialIssuerService() {
+        var configurationService = new ConfigurationService();
+        dataStore = getDataStore(configurationService);
+    }
+
+    public CredentialIssuerService(DataStore<AddressSessionItem> dataStore) {
+        this.dataStore = dataStore;
+    }
+
+    public UUID getSessionId(APIGatewayProxyRequestEvent input)
+            throws CredentialRequestException, DynamoDbException {
+        var queryParams = queryParams(input.getBody());
+        if (!queryParams.containsKey(SUB)) {
+            throw new CredentialRequestException(ErrorResponse.INVALID_REQUEST_PARAM);
+        }
+        var accessToken = getAccessToken(input.getHeaders());
+        var addressSessionTable = dataStore.getTable();
+        var index = addressSessionTable.index(AddressSessionItem.ACCESS_TOKEN_INDEX);
+
+        var listHelper = new ListUtil();
+        var addressSessionItem =
+                listHelper.getValueOrThrow(dataStore.getItemByGsi(index, accessToken));
+
+        return addressSessionItem.getSessionId();
+    }
+
+    public List<CanonicalAddressWithResidency> getAddresses(UUID sessionId)
+            throws CredentialRequestException {
+        var addressSessionItem = dataStore.getItem(sessionId.toString());
+        if (addressSessionItem == null) {
+            throw new CredentialRequestException(ErrorResponse.MISSING_ADDRESS_SESSION_ITEM);
+        }
+        return addressSessionItem.getAddresses();
+    }
+
+    private String getAccessToken(Map<String, String> headers) throws CredentialRequestException {
```

Comment:
After reading https://github.com/alphagov/digital-identity-architecture/blob/main/rfc/0001-crossteam-oauth-profile.md#requests-to-the-protected-resource, its clear that IPV Core MUST send bearer tokens passed in the Authentication header, see [RFC6750](https://datatracker.ietf.org/doc/html/rfc6750). So this check may need some rework, sorry!